### DISCUSSION TITLE
'Installer failed with exit code: 0x80070002 : The system cannot find the file specified.' when updating AppInstaller package

### DISCUSSION THREAD

**Discussion by shockogit** [microsoft/winget-cli#4766] [Q&A]
Our IT sec team block the usage of MS Store UI via group policy. Many of our machines do not have the AppInstaller package nor winget so I have been looking at installing manually to test and therafter looking to distibute via MECM. I'm a local administrator so I  install the MSIXBundle package as follows offline:

`PS > Add-AppxPackage -path 'Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'`

It installs fine and I cna now use Winget. However, this verison of the AppInstaller is somewhat old

`PS > (Get-AppxPackage -Name Microsoft.DesktopAppInstaller).Version
1.22.11132.0`

So naturally I run `winget update Microsoft.AppInstaller` but I get the following:

`Found App Installer [Microsoft.AppInstaller] Version 1.23.1911.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
  - Packages
      Microsoft.UI.Xaml.2.8 [>= 8.2310.30001.0]
Successfully verified installer hash
Starting package install...
Installer failed with exit code: 0x80070002 : The system cannot find the file specified.`

Why is this?

**Comment by denelon** [comment]
The App Installer has UI components which is why there is a dependency on UI.XAML.

You may have better results using `Repair-WinGetPackageManager -Latest -Force` in the Microsoft.WinGet.Client PowerShell module.

Note: Repair works in Windows PowerShell but several of the other cmdlets do not work in Windows PowerShell.
